apis/compute/v1beta1/gkecluster_types.go

```diff
@@ -0,0 +1,1346 @@
+	// PrivateEndpoint: Output only. The internal IP address of this
+	// cluster's master endpoint.
+	PrivateEndpoint string `json:"privateEndpoint,omitempty"`
+
+	// PublicEndpoint: Output only. The external IP address of this
+	// cluster's master endpoint.
+	PublicEndpoint string `json:"publicEndpoint,omitempty"`
+}
+
+// ResourceUsageExportConfig is configuration for exporting cluster
+// resource usages.
+type ResourceUsageExportConfig struct {
+	// BigqueryDestination: Configuration to use BigQuery as usage export
+	// destination.
+	// +optional
+	BigqueryDestination *BigQueryDestination `json:"bigqueryDestination,omitempty"`
+
+	// ConsumptionMeteringConfig: Configuration to enable resource
+	// consumption metering.
+	// +optional
+	ConsumptionMeteringConfig *ConsumptionMeteringConfig `json:"consumptionMeteringConfig,omitempty"`
+
+	// EnableNetworkEgressMetering: Whether to enable network egress
+	// metering for this cluster. If enabled, a
+	// daemonset will be created in the cluster to meter network egress
+	// traffic.
+	// +optional
+	EnableNetworkEgressMetering *bool `json:"enableNetworkEgressMetering,omitempty"`
+}
+
+// BigQueryDestination is parameters for using BigQuery as the destination
+// of resource usage export.
+type BigQueryDestination struct {
+	// DatasetId: The ID of a BigQuery Dataset.
+	DatasetID string `json:"datasetId,omitempty"`
+}
+
+// ConsumptionMeteringConfig is parameters for controlling consumption
+// metering.
+type ConsumptionMeteringConfig struct {
+	// Enabled: Whether to enable consumption metering for this cluster. If
+	// enabled, a
+	// second BigQuery table will be created to hold resource
+	// consumption
+	// records.
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// TierSettings is cluster tier settings.
+type TierSettings struct {
+	// Tier: Cluster tier.
+	//
+	// Possible values:
+	//   "UNSPECIFIED" - UNSPECIFIED is the default value. If this value is
+	// set during create or
+	// update, it defaults to the project level tier setting.
+	//   "STANDARD" - Represents the standard tier or base Google Kubernetes
+	// Engine offering.
+	//   "ADVANCED" - Represents the advanced tier.
+	Tier string `json:"tier,omitempty"`
+}
+
+// VerticalPodAutoscaling contains global,
+// per-cluster information
+// required by Vertical Pod Autoscaler to automatically adjust
+// the resources of pods controlled by it.
+type VerticalPodAutoscaling struct {
+	// Enabled: Enables vertical pod autoscaling.
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// WorkloadIdentityConfig is configuration for the use of Kubernetes
+// Service Accounts in GCP IAM
+// policies.
+type WorkloadIdentityConfig struct {
+	// IdentityNamespace: IAM Identity Namespace to attach all Kubernetes
+	// Service Accounts to.
+	IdentityNamespace string `json:"identityNamespace,omitempty"`
+}
+
+// NodePoolClusterStatus is a subset of information about NodePools associated
+// with a GKE cluster.
+type NodePoolClusterStatus struct {
```

Comment:
This is a subset of the fields exposed for `NodePools` on a `GKECluster`. It could be argued that this is not "high fidelity", but since this is only present in status, a subset seems fine to me. It is the same fields that will be present in our `NodePoolObervation` struct + `Name` and `Version`

Reply:
What's the thinking around only exposing a subset rather than all fields?

Reply:
Mostly just because we don't allow for configuration of fields here and so the primary objective seems to just be able to identify that a node pool exists and see its status. It also would add quite a bit of duplicate code in `NodePool` and `GKECluster`.